components/skus/browser/rs/cxx/src/httpclient.rs

```diff
@@ -50,9 +50,9 @@ impl From<ffi::HttpResponse<'_>> for Result<http::Response<Vec<u8>>, InternalErr
     fn from(resp: ffi::HttpResponse<'_>) -> Self {
         match resp.result {
             ffi::SkusResult::Ok => {
-                let mut response = http::Response::builder();
+                let mut binding = http::Response::builder();
+                let mut response = binding.status(resp.return_code);
```

Comment:
Aha. I think that's because it's building against `http v0.1` (as vendored) instead of the specifed `http v0.2` where the api is slightly different. Maybe try it after addressing https://github.com/brave/brave-core/pull/20139#discussion_r1324800901?